components/lock/redis/cluster_redis_lock.go

```diff
@@ -0,0 +1,243 @@
+//
+// Copyright 2021 Layotto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package redis
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-redis/redis/v8"
+	"mosn.io/layotto/components/lock"
+	"mosn.io/layotto/components/pkg/utils"
+	msync "mosn.io/mosn/pkg/sync"
+	"mosn.io/pkg/log"
+	"strings"
+	"sync"
+	"time"
+)
+
+//RedLock
+//it will be best to use at least 5 hosts
+type ClusterRedisLock struct {
+	clients  []*redis.Client
+	metadata utils.RedisClusterMetadata
+	workpool msync.WorkerPool
+	replicas int
```

Comment:
unused replicas?

Reply:
Removed.
It's because I added a unused `replicas`field in the redis standalone lock